mas_arena/agents/autogen.py

```diff
@@ -0,0 +1,129 @@
+import os
+from typing import Dict, Any, List
+import contextlib
+# from openai import OpenAI
+from openai import AsyncOpenAI
+from dotenv import load_dotenv
+from mas_arena.agents.base import AgentSystem, AgentSystemRegistry
+
+# Load environment variables
+load_dotenv()
+
+class AutoGen(AgentSystem):
+    """
+    AutoGen System
+
+    This agent system implements a collaborative AutoGen framework similar to AutoGen,
+    where AutoGen agents with distinct roles interact to solve problems through conversation.
+    """
+    def __init__(self, name: str = "autogen", config: Dict[str, Any] = None):
+        """Initialize the AutoGen System"""
+        super().__init__(name, config)
+        self.config = config or {}
+        
+        # Default model and agent configurations
+        self.model_name = self.config.get("model_name") or os.getenv("MODEL_NAME", "qwen-plus")
+        self.system_prompt = self.config.get("system_prompt", "") + self.format_prompt
+        
+        # Initialize OpenAI client
+        # self.client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"), base_url=os.getenv("OPENAI_API_BASE"))
+        self.client = AsyncOpenAI(api_key=os.getenv("OPENAI_API_KEY"), base_url=os.getenv("OPENAI_API_BASE"))
+        # Define AutoGen agents with distinct roles
+        self.agents = self.config.get("agents", [
```

Comment:
This implementation is not compatible with the tool usage. Please check the `extending.md`.